src/invidious/yt_backend/extractors.cr

```diff
@@ -118,12 +125,25 @@ private module Parsers
         case b["label"].as_s
         when "LIVE NOW"
           live_now = true
-        when "New", "4K", "CC"
-          # TODO
+        when "New"
+          is_new = true
+        when "4K"
+          is_4k = true
+        when "8K"
+          is_8k = true
+        when "VR180"
+          is_vr180 = true
+        when "360°"
+          is_vr360 = true
+        when "3D"
+          is_3d = true
+        when "CC"
+          has_captions = true
         when "Premium"
           # TODO: Potentially available as item_contents["topStandaloneBadge"]["metadataBadgeRenderer"]
           premium = true
```

Comment:
Can you refactor all of these as a "Flags enum" please?
https://crystal-lang.org/api/1.12.2/Flags.html

Something like that (note that `Enum` members must start with a letter):
```
@[Flags]
enum VideoBadges
  LiveNow
  Premium
  ThreeD
  FourK
end
```

then:
```
badges = VideoBadges::None

case xxx
when "4K"
  badges |= VideoBadges::FourK
end
```

And finally:
```
json.field "is4k", badges.four_k?
```